File: android/app/src/main/kotlin/com/thecityandthebike/ui/components/LoadingIndicator.kt
package com.thecityandthebike.ui.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableIntStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import kotlinx.coroutines.delay

private val flavorTexts = listOf(
    "Waking up servers…",
    "Sending to space…",
    "Tagging…",
    "Jiggling electrons…",
    "Popping a wheelie…",
    "Paying the piper…",
    "Cropping…",
)

@Composable
fun LoadingIndicator(modifier: Modifier = Modifier) {
    var index by remember { mutableIntStateOf(0) }

    LaunchedEffect(Unit) {
        while (true) {
            delay(2000)
            index = (index + 1) % flavorTexts.size
        }
    }

    Column(
        modifier = modifier,
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        CircularProgressIndicator()
        Spacer(modifier = Modifier.height(16.dp))
        Text(
            text = flavorTexts[index],
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )
    }
}

File: ios/Packages/TCATBFeed/Sources/TCATBFeed/Components/LoadingIndicator.swift
import SwiftUI

private let flavorTexts = [
    "Waking up servers\u{2026}",
    "Sending to space\u{2026}",
    "Tagging\u{2026}",
    "Jiggling electrons\u{2026}",
    "Popping a wheelie\u{2026}",
    "Paying the piper\u{2026}",
    "Cropping\u{2026}",
]

public struct LoadingIndicator: View {
    @State private var index = 0

    public init() {}

    public var body: some View {
        VStack(spacing: 16) {
            ProgressView()
            Text(flavorTexts[index])
                .font(.subheadline)
                .foregroundStyle(.secondary)
        }
        .task {
            while !Task.isCancelled {
                try? await Task.sleep(for: .seconds(2))
                index = (index + 1) % flavorTexts.count
            }
        }
    }
}
